service/matching/taskReader.go

```diff
@@ -394,19 +395,28 @@ func (tr *taskReader) dispatchSingleTaskFromBuffer(isolationGroup string, taskIn
 	err := tr.dispatchTask(dispatchCtx, task)
 	timerScope.Stop()
 	cancel()
+
 	if err == nil {
 		return false, true
 	}
-	if err == context.Canceled {
+
+	if errors.Is(err, context.Canceled) {
 		tr.logger.Info("Tasklist manager context is cancelled, shutting down")
 		return true, true
 	}
-	if err == context.DeadlineExceeded {
+
+	if errors.Is(err, context.DeadlineExceeded) {
```

Comment:
👍 seems very likely safe and correct.  code here doesn't seem to wrap anything ever, but this might allow us to.